src/core/playground/wrapped.ts

```diff
@@ -0,0 +1,78 @@
+import type { Deserializer, Serializer } from "../../bcs";
+import { HexInput, SigningScheme as AuthenticationKeyScheme } from "../../types";
+import { AuthenticationKey } from "./authenticationKey";
+import { bcsToBytes } from "./bcs";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import type { PublicKey } from "./interfaces";
+import { SignatureScheme } from "./scheme";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+
+export type AllowedSignatures = Ed25519Signature | Secp256k1Signature;
+
+export class WrappedSignature<TSignature extends AllowedSignatures = AllowedSignatures> {
```

Comment:
tbh, I prefer `AnySignature` over `WrappedSignature`, I also like `AllowedSignature`

Reply:
Fair, it would be good to be consistent with the rust SDK anyway.
Will revert